File: BatWoman-backend/src/main/java/com/BatWoman/BatWoman_backend/service/WishlistService.java
package com.BatWoman.BatWoman_backend.service;

import com.BatWoman.BatWoman_backend.dto.wishlist.WishlistResponse;

import java.util.List;
import java.util.UUID;

public interface WishlistService {

    List<WishlistResponse> getMyWishlist();

    WishlistResponse addToWishlist(UUID productId);

    void removeFromWishlist(UUID productId);

    boolean isInWishlist(UUID productId);
}

File: BatWoman-frontend/batwoman-frontend/services/wishlist.service.ts
import api from "@/lib/axios";

import { WishlistResponse } from "@/types/wishlist";

export async function getWishlist(): Promise<WishlistResponse[]> {
  const { data } = await api.get<WishlistResponse[]>(
    "/wishlist"
  );

  return data;
}

export async function addToWishlist(
  productId: string
): Promise<WishlistResponse> {
  const { data } = await api.post<WishlistResponse>(
    `/wishlist/${productId}`
  );

  return data;
}

export async function removeFromWishlist(
  productId: string
): Promise<void> {
  await api.delete(`/wishlist/${productId}`);
}

export async function isInWishlist(
  productId: string
): Promise<boolean> {
  const { data } = await api.get<boolean>(
    `/wishlist/${productId}/exists`
  );

  return data;
}